units/recalculate_change_points.go

```diff
@@ -0,0 +1,243 @@
+	env            cedar.Environment
+	metricGrouping MetricGrouping
+}
+
+func init() {
+	registry.AddJobType("recalculate-change-points", func() amboy.Job { return makeChangePointsJob() })
+}
+
+func makeChangePointsJob() *recalculateChangePointsJob {
+	j := &recalculateChangePointsJob{
+		Base: &job.Base{
+			JobType: amboy.JobType{
+				Name:    "recalculate-change-points",
+				Version: 1,
+			},
+		},
+	}
+	j.SetDependency(dependency.NewAlways())
+	return j
+}
+
+func NewRecalculateChangePointsJob(metricGrouping MetricGrouping) amboy.Job {
+	j := makeChangePointsJob()
+	// Every ten minutes at most
+	timestamp := util.RoundPartOfHour(10)
+	j.SetID(fmt.Sprintf("%s.%s.%s.%s.%s.%s", j.JobType.Name, metricGrouping.Id.Project, metricGrouping.Id.Variant, metricGrouping.Id.Task, metricGrouping.Id.Test, timestamp))
+	j.metricGrouping = metricGrouping
+	return j
+}
+
+type MetricGrouping struct {
+	Id struct {
+		Project string `bson:"project"`
+		Variant string `bson:"variant"`
+		Task    string `bson:"task"`
+		Test    string `bson:"test"`
+	} `bson:"_id"`
+}
+
+func makeSeriesFilter(grouping MetricGrouping) bson.M {
+	return bson.M{
+		"info.project":   grouping.Id.Project,
+		"info.variant":   grouping.Id.Variant,
+		"info.task_name": grouping.Id.Task,
+		"info.test_name": grouping.Id.Test,
+	}
+
+}
+
+func makeTimeSeriesPipeline(grouping MetricGrouping) []bson.M {
+	var pipeline []bson.M
+
+	pipeline = append(pipeline, bson.M{
+		"match": makeSeriesFilter(grouping),
+	})
+
+	pipeline = append(pipeline, bson.M{
+		"$match": bson.M{
+			"info.order": bson.M{
+				"$exists": true,
+			},
+			"info.mainline": true,
+		},
+	})
+
+	pipeline = append(pipeline, bson.M{
+		"$unwind": "$rollups.stats",
+	})
+
+	pipeline = append(pipeline, bson.M{
+		"$project": bson.M{
+			"measurement": "$rollups.stats.name",
+			"value":       "$rollups.stats.val",
+			"order":       "$info.order",
+		},
+	})
+
+	pipeline = append(pipeline, bson.M{
+		"$group": bson.M{
+			"_id": "$measurement",
+			"time_series": bson.M{
+				"$push": bson.M{
+					"id":    "$_id",
+					"value": "$value",
+					"order": "$order",
+				},
+			},
+		},
+	})
+
+	return pipeline
+}
+
+type errorMsg = map[string]interface{}
+
+type timeSeriesElement struct {
+	PerfResultID string `bson:"id"`
+	Value        float64
+	Order        int
+}
+
+type measurementTimeSeries struct {
+	Measurement string              `bson:"_id"`
+	TimeSeries  []timeSeriesElement `bson:"times_series"`
+}
+
+func (j *recalculateChangePointsJob) Run(ctx context.Context) {
+	defer j.MarkComplete()
+	if j.env == nil {
+		j.env = cedar.GetEnvironment()
+	}
+	conf := model.NewCedarConfig(j.env)
+	err := conf.Find()
+	if err != nil {
+		grip.Error(errorMsg{
+			"message": "Unable to get cedar configuration",
+			"error":   err,
+		})
+	}
+	detector := perf.NewMicroServiceChangeDetector(conf.ChangeDetector.URI, conf.ChangeDetector.User, conf.ChangeDetector.Token)
+	collection := j.env.GetDB().Collection("perf_results")
```

Comment:
put this query in the models package. also use constants in the model package for results.